File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        if (n == 2) return {0, 1};

        unordered_map <int, int> mp;
        for (int i = 0; i < n; i++) {
            int comp = target - nums[i];
            if (mp.find(comp) != mp.end()) 
                return {mp[comp], i};
            else 
                mp[nums[i]] = i;
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map <Integer, Integer> HM = new HashMap <> ();
        
        for (int i = 0; i< nums.length; i++){
            int compliment = target - nums[i];
            
            if (HM.containsKey(compliment)) {
                return new int[]{i, HM.get(compliment)};
            } else {
                HM.put(nums[i], i);
            }
        }
        return null;
    }
}